src/box/lua/trigger.c

```diff
@@ -226,7 +346,9 @@ luaT_trigger_pairs(struct lua_State *L)
 		lua_pushcfunction(L, luaT_trigger_iterator_next_exhausted);
 		return 1;
 	}
-	lua_pushcfunction(L, luaT_trigger_iterator_next);
+	size_t *epoch = lua_newuserdata(L, sizeof(size_t));
```

Comment:
I don't understand why we need to pass the "epoch" as a Lua object. I think we should create a weak reference object for functions, like we did for index, see [`index_weak_ref.h`](https://github.com/tarantool/tarantool/blob/58f5ffe43bffa0901d8d577bbb4d523c4e7e682a/src/box/index_weak_ref.h).

Reply:
Discussed f2f, this part was reworked.